cmd/golangBuild.go

```diff
@@ -540,33 +527,48 @@ func runGolangBuildPerArchitecture(config *golangBuildOptions, goModFile *modfil
 }
 
 // lookupPrivateModulesRepositories returns a slice of all modules that match the given glob pattern
-func lookupGolangPrivateModulesRepositories(goModFile *modfile.File, globPattern string, utils golangBuildUtils) ([]string, error) {
+func lookupGolangPrivateModulesRepositories(goModFile *modfile.File, globPattern string, token string, utils golangBuildUtils) error {
 	if globPattern == "" {
-		return []string{}, nil
+		return nil
 	}
 
 	if goModFile == nil {
-		return nil, fmt.Errorf("couldn't find go.mod file")
+		return fmt.Errorf("couldn't find go.mod file")
 	} else if goModFile.Require == nil {
-		return []string{}, nil // no modules referenced, nothing to do
+		return nil // no modules referenced, nothing to do
 	}
 
-	privateModules := []string{}
-
 	for _, goModule := range goModFile.Require {
 		if !module.MatchPrefixPatterns(globPattern, goModule.Mod.Path) {
+
 			continue
 		}
 
 		repo, err := utils.GetRepositoryURL(goModule.Mod.Path)
 
 		if err != nil {
-			return nil, err
+			return err
+		}
+
+		parsedRepoURL, err := url.Parse(repo)
+		if err != nil {
+			return err
+		}
+		repoBaseURL := fmt.Sprintf("%s://%s", parsedRepoURL.Scheme, parsedRepoURL.Host)
```

Comment:
There is no check if the host URL already occurred before, resulting in multiple calls of `git config` with the same host URL. It would be neat if only unique host URLs are considered. But it's not strictly necessary.